thali/thalicryptomanager.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+var crypto = require('crypto');
+var fs = require('fs');
+var path = require('path');
+
+// The password is not secure because anyone who can get to the file can get
+// to the app and thus can get the password. The password is used here only to
+// satisfy the crypto/PKCS12 APIs.
+var password = 'password';
+var certname = 'certname';
+var country = 'country';
+var organization = 'organization';
+var pkcs12FileName = '/pkcs12.pfx';
+var macName = 'SHA256';
+var hashSizeInBytes = 16;
+
+module.exports = {
```

Comment:
Although this use of module.exports is technically correct it is not idiomatic. When a module exports multiple functions this is usually handled by having each function individually declared using exports.x = function() (e.g. look at the Exporting a module section of http://www.sitepoint.com/understanding-module-exports-exports-node-js/). Please update accordingly.

The reason for this, btw, is to make it easy to intersperse exported code with non-exports content like purely local functions.